feat(widget-builder): Add GA documentation - [DD-539] (#5065)
getsentry/sentry-docs: src/docs/product/dashboards/widget-builder/index.mdx

---
title: Widget Builder

> sidebar_order: 20

Author: I'm not sure about this number. Can it be any? @imatwawana

> sidebar_order: 20

Reviewer: This just needs to be between the number on the Custom Dashboards page (10) and the one on the Widget Library page (20). 

```suggestion
sidebar_order: 15
```

description: "Learn how to create widgets for your dashboards or edit existing ones."
---

<Note>

This feature is available only if your organization is on either a Business or Trial plan.

</Note>

> When adding a widget or editing an existing one, you will be taken to a new page where you can see the dashboard widget builder, as shown below:

Reviewer: ```suggestion
When adding a widget or editing an existing one, you will be taken to a new page where you can see the dashboard widget builder:
```

- Line charts are a general-purpose visualization. Make sure the Y-Axis units match if you add multiple overlays. Mixing count with duration will not yield useful results.

### Table

Table charts display the results of a [Discover Query](/product/discover-queries/). This visualization is well suited to showing key fields and related aggregates. For example “duration percentiles for the most frequently visited transaction names”. You can also choose to display [issues](/product/issues/) data in a table visualization using the [dataset selector](/product/dashboards/widget-builder/#choose-your-dataset).

### World Map

Events are queried using the given function and `geo.country_code`. The results are plotted as a density value on a world map. An example scenario would be “in which countries are users experiencing the most errors”. You can zoom the map using the mouse wheel, or pan the map by clicking and dragging the map around.

### Big Number

A big number visualization displays the current value of a single function. This visualization is well suited for high-level aggregates. An example scenario would be “p95 of all transactions”.

## Choose your dataset

> In the dataset selection step, you can choose which type of data you would like to use in your widget. This data is classified into three different datasets:

Reviewer: ```suggestion
In the dataset selection step, choose which type of data you would like to use in your widget. This data is classified into two different datasets: [errors and transactions](#errors--transactions) and [issues](#issues).
```

Author: good catch! the UI will not have releases for now

## Choose your columns

The option to set columns is only available for [table visualizations](/product/dashboards/widget-builder/#table).

By default, the first column is already configured for you, but you can update it. To add more columns, which stacks events, add [functions](/product/discover-queries/query-builder/#stacking-functions) (f(x)) that can take in parameters. [Field and tag columns](/product/sentry-basics/search/searchable-properties/?_ga=2.209391633.2075508021.1652792062-1379607348.1646725525#event-properties) will help you view more details about the events (for example, title).

You can also add equations to your query using the table columns as equation variables, and the results will be displayed in the query results table. Learn more in [Adding Query Equations](/product/discover-queries/query-builder/query-equations/).

Next to each column and equation, you have the option to set an alias that will replace the column name in the table header. 

If two or more columns are added, you can change the order in which they're displayed in the table by clicking and dragging them:

![Drag and drop column.](widget-builder-drag-and-drop-column.png)

> ## Choose what to plot in the y-axis

Reviewer: ```suggestion
## Choose What to Plot in the Y-axis
```

The option to set a y-axis is available on [all visualizations](/product/dashboards/widget-builder/#choose-your-visualization) except [table](/product/dashboards/widget-builder/#table). In the case of [time-series visualizations](/product/dashboards/widget-builder/#area-chart--bar-chart--line-chart), you'll be able to plot up to three values that can be functions or equations. If the units of your selections conflict, then the y-axis of the chart will be labeled using the units of the first selection.

You can also choose to construct an equation by clicking “Add an Equation”. Doing this displays a text box where you can enter your equation. Functions that you've added previously in this step will appear for autocompletion, but you can add other functions as well.

## Filter your results

> Use this field to filter your results by the searchable properties of your [dataset selection](/product/dashboards/widget-builder/#choose-your-dataset). For more information on which properties apply to each dataset, check out our [full searchable properties documentation](/product/sentry-basics/search/searchable-properties/#properties-table).

Reviewer: ```suggestion
Use this field to filter your results by the searchable properties of your [dataset selection](/product/dashboards/widget-builder/#choose-your-dataset). For more information on which properties apply to each dataset, check out our [full Searchable Properties documentation](/product/sentry-basics/search/searchable-properties/#properties-table).
```

> On [time-series visualizations](/product/dashboards/widget-builder/#area-chart--bar-chart--line-chart) you can add up to three queries to filter your results, each with their own search conditions to compare results. Each query will generate a new series in your chart for each y-axis. To make it easier to differentiate between them in the legend and tooltip, you can add a legend alias that will be added as a prefix to the labels.

Reviewer: We should definitely add a screenshot after this paragraph--this is pretty hard to visualize.

```suggestion
On [time-series visualizations](/product/dashboards/widget-builder/#area-chart--bar-chart--line-chart) you can add up to three queries to filter your results, each with their own search conditions, to compare results. Each query will generate a new series in your chart for each y-axis. To make it easier to differentiate between them in the legend and tooltip, you can add a legend alias that will be added as a prefix to the labels.
```

> ## Group your results

Reviewer: ```suggestion
## Group Your Results
```

The option to group results is only available for [time-series visualizations](/product/dashboards/widget-builder/#area-chart--bar-chart--line-chart). You can group your time series results by different fields or tags such as project, environment, or transaction. You can select up to 20 fields to group your data.

> If two or more groups are added, a draggable element will be displayed that you can use to sort the fields, but the sort does not impact how your data is displayed in the visualization. 

Reviewer: ```suggestion
If two or more groups are added, a draggable element will be displayed that you can use to sort the fields, but the sort does not impact how your data is displayed in the visualization:
```

![Drag and drop group by.](widget-builder-drag-and-drop-group-by.png)

> ## Sort by a column

Reviewer: ```suggestion
## Sort by a Column
```

> The option to sort by a column is only available for [table visualizations](/product/dashboards/widget-builder/#table). Choose one of the columns of your table to sort by and set whether you want to sort high to low or low to high.

Reviewer: ```suggestion
The option to sort by a column is only available for [table visualizations](/product/dashboards/widget-builder/#table). Choose one of the columns of your table to sort by, and set whether you want to sort high-to-low or low-to-high.
```

## Limit

The option to limit your results is only available for [time-series visualizations](/product/dashboards/widget-builder/#area-chart--bar-chart--line-chart) where you've applied a grouping. Depending on your configuration (for example, multiple y-axes, multiple queries), your chart results could become unmanageable, so a limit is imposed. You can change this limit by selecting it in the dropdown. We intend to show at most 10 series (that is, lines in a line chart or bars in a bar chart) when multiple y-axes or queries are selected.

> ## Sort by a y-axis

Reviewer: ```suggestion
## Sort by a Y-axis
```

> The option to sort by a y-axis is only available if you've applied a grouping to your data (in [“Group your results”](/product/dashboards/widget-builder/#group-your-results)). The sorting applies to the underlying groups in your results and thus affects the series that appear in your chart. You can select fields that were chosen in the group by section, select equations defined in the y-axes, construct any function similar to how you would select a function for plotting, or construct a custom equation.

Reviewer: ```suggestion
The option to sort by a y-axis is only available if you've applied a grouping to your data (in [“Group your results”](/product/dashboards/widget-builder/#group-your-results)). The sorting applies to the underlying groups in your results and thus affects the series that appear in your chart. You can select fields that you used for grouping, select equations defined in the y-axes, construct any function similar to how you would select a function for plotting, or construct a custom equation.
```

> If you select “Custom Equation”, an extra field appears where you can enter an equation of your choice that will be applied to the query.

Reviewer: ```suggestion
If you select “Custom Equation”, an extra field appears where you can enter an equation of your choice that will be applied to the query:
```

---
title: Widget Builder
sidebar_order: 15
description: "Learn how to create widgets for your dashboards or edit existing ones."
---

<Note>

This feature is available only if your organization is on either a Business or Trial plan.

</Note>

> 

Reviewer: ```suggestion

When adding a widget or editing an existing one, you will be taken to a new page where you can see the dashboard widget builder:

```

### Errors & Transactions

Choosing "Errors and Transactions" allows you to query and aggregate error and transaction events in the same way you would for a [Discover Query](/product/discover-queries/). Some widget examples include: 

- Tracking performance of an endpoint 
- Throughput by country 
- Users most affected by errors

### Issues
Choosing "Issues (States, Assignment, Time, etc.)" allows you to query issue properties, such as `is:unresolved` or event properties such as `error.handled:0`, and to sort by issue fields such as `First Seen`. When searching on event properties, the search will return any issue that has one or more events matching the supplied event filters. 
Choose this dataset if you want to customize a list of issues on your dashboard. Some widget examples include:

- Issues assigned to your team 
- Most frequently occurring issues
- The "Issues" dataset is only available in table visualization widgets and its disabled if you have other visualizations selected.

## Choose Your Columns

The option to set columns is only available for [table visualizations](/product/dashboards/widget-builder/#table).

> By default, the first column is already configured for you, but you can update it. To add more columns, which stacks events, add [functions](/product/discover-queries/query-builder/#stacking-functions) (`f(x)`) that can take in parameters. [Field and tag columns](/product/sentry-basics/search/searchable-properties/?_ga=2.209391633.2075508021.1652792062-1379607348.1646725525#event-properties) will help you view more details about the events (for example, title).

Reviewer: Let's add at least one more example of a field or tag column in the parentheses at the end of this example.

![Drag and drop column.](widget-builder-drag-and-drop-column.png)

## Choose What to Plot in the Y-axis

The option to set a y-axis is available on all visualizations except [table](/product/dashboards/widget-builder/#table). In the case of [time-series visualizations](/product/dashboards/widget-builder/#area-chart--bar-chart--line-chart), you'll be able to plot up to three values that can be functions or equations. If the units of your selections conflict, then the y-axis of the chart will be labeled using the units of the first selection.

You can also choose to construct an equation by clicking “Add an Equation”. Doing this displays a text box where you can enter your equation. Functions that you've added previously in this step will appear for autocompletion, but you can add other functions as well.

## Filter Your Results

Use this field to filter your results by the searchable properties of your [dataset selection](/product/dashboards/widget-builder/#choose-your-dataset). For more information on which properties apply to each dataset, check out our [full Searchable Properties documentation](/product/sentry-basics/search/searchable-properties/#properties-table).

> On [time-series visualizations](/product/dashboards/widget-builder/#area-chart--bar-chart--line-chart) you can add up to three queries to filter your results, each with their own search conditions, to compare results. Each query will generate a new series in your chart for each y-axis. To make it easier to differentiate between them in the legend and tooltip, you can add a legend alias that will be added as a prefix to the labels.

Reviewer: Let's add a screenshot after this paragraph to make this easier to understand.

Author: I've added a screenshot after this paragraph. Please let me know what you think

![image](https://user-images.githubusercontent.com/29228205/170946693-80680ad5-8815-4fed-9437-b2c4da576122.png)

### Errors & Transactions

Choosing "Errors and Transactions" allows you to query and aggregate error and transaction events in the same way you would for a [Discover Query](/product/discover-queries/). Some widget examples include: 

- Tracking performance of an endpoint 
- Throughput by country 
- Users most affected by errors

### Issues
Choosing "Issues (States, Assignment, Time, etc.)" allows you to query issue properties, such as `is:unresolved` or event properties such as `error.handled:0`, and to sort by issue fields such as `First Seen`. When searching on event properties, the search will return any issue that has one or more events matching the supplied event filters. 
Choose this dataset if you want to customize a list of issues on your dashboard. Some widget examples include:

- Issues assigned to your team 
- Most frequently occurring issues
- The "Issues" dataset is only available in table visualization widgets and its disabled if you have other visualizations selected.

## Choose Your Columns

The option to set columns is only available for [table visualizations](/product/dashboards/widget-builder/#table).

> By default, the first column is already configured for you, but you can update it. To add more columns, which stacks events, add [functions](/product/discover-queries/query-builder/#stacking-functions) (`f(x)`) that can take in parameters. [Field and tag columns](/product/sentry-basics/search/searchable-properties/?_ga=2.209391633.2075508021.1652792062-1379607348.1646725525#event-properties) will help you view more details about the events (for example, title).

Reviewer: ```suggestion
By default, the first column is already configured for you, but you can update it. To add more columns, which stacks events, add [functions](/product/discover-queries/query-builder/#stacking-functions) (`f(x)`) that can take in parameters. [Field and tag columns](/product/sentry-basics/search/searchable-properties/#event-properties) will help you view more details about the events (for example, title).
```

### Errors & Transactions

Choosing "Errors and Transactions" allows you to query and aggregate error and transaction events in the same way you would for a [Discover Query](/product/discover-queries/). Some widget examples include: 

- Tracking performance of an endpoint 
- Throughput by country 
- Users most affected by errors

### Issues
Choosing "Issues (States, Assignment, Time, etc.)" allows you to query issue properties, such as `is:unresolved` or event properties such as `error.handled:0`, and to sort by issue fields such as `First Seen`. When searching on event properties, the search will return any issue that has one or more events matching the supplied event filters. 
Choose this dataset if you want to customize a list of issues on your dashboard. Some widget examples include:

- Issues assigned to your team 
- Most frequently occurring issues
- The "Issues" dataset is only available in table visualization widgets and its disabled if you have other visualizations selected.

## Choose Your Columns

The option to set columns is only available for [table visualizations](/product/dashboards/widget-builder/#table).

> By default, the first column is already configured for you, but you can update it. To add more columns, which stacks events, add [functions](/product/discover-queries/query-builder/#stacking-functions) (`f(x)`) that can take in parameters. [Field and tag columns](/product/sentry-basics/search/searchable-properties/#event-properties) will help you view more details about the events (for example, `title` to see the description of an issue and `users` to see how many users it's affecting).

Reviewer: ```suggestion
By default, the first column is already configured for you, but you can update it. To add more columns, which stacks events, add [functions](/product/discover-queries/query-builder/#stacking-functions) (`f(x)`) that can take in parameters. [Field and tag columns](/product/sentry-basics/search/searchable-properties/#event-properties) will help you view more details about the events (for example, `title` to see the description of an issue or `users` to see how many users it's affecting).
```

You can also choose to construct an equation by clicking “Add an Equation”. Doing this displays a text box where you can enter your equation. Functions that you've added previously in this step will appear for autocompletion, but you can add other functions as well.

## Filter Your Results

Use this field to filter your results by the searchable properties of your [dataset selection](/product/dashboards/widget-builder/#choose-your-dataset). For more information on which properties apply to each dataset, check out our [full Searchable Properties documentation](/product/sentry-basics/search/searchable-properties/#properties-table).

On [time-series visualizations](/product/dashboards/widget-builder/#area-chart--bar-chart--line-chart) you can add up to three queries to filter your results, each with their own search conditions, to compare results. Each query will generate a new series in your chart for each y-axis. To make it easier to differentiate between them in the legend and tooltip, you can add a legend alias that will be added as a prefix to the labels, as in the example below:

> ![Filter your results.](widget-builder-filter-your-results-with-three-queries.png)

Reviewer: This is great! Let's crop this image a bit to make it easier to see the parts we're explaining; let's crop out the right sidebar entirely and the "Group your results" section since it's not relevant to the example.
